README.md

```diff
@@ -110,3 +110,19 @@ following config in config.yml.
 SilverStripe\ExternalLinks\Tasks\CurlLinkChecker::
   bypass_cache: 1
 ```
+
+## Headers
+
+You may want to set headers to be sent with the CURL request (eg: user-agent) to avoid website rejecting the request thinking it is a bot.
+You can set them with the following config in config.yml.
+
+```yaml
+# Headers
+SilverStripe\ExternalLinks\Tasks\CurlLinkChecker:
+  user_agent: "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:53.0) Gecko/20100101 Firefox/53.0"
+  headers:
+    - "accept-encoding: gzip, deflate, br"
+    - "referer: https://www.domain.com/"
+    - "sec-fetch-mode: navigate"
```

Comment:
This makes it very difficult to remove headers, say if this module decides at some point to configure some headers by default. If you use keys then it will be easy to do so, i.e.:

```yaml
headers:
  Accept-Encoding: gzip, deflate, br
  Referrer: https://www.domain.com
  ...
```

You'd need to join the keys and value before giving them to cURL too

Reply:
You are right to say that having the keys is better. I've removed the user_agent config so it is clear it needs to be added in the header array.